### DISCUSSION TITLE
RESOLVED: does zwave-js-ui support 'tcp' serial port connections?

### DISCUSSION THREAD

**Discussion by ltgcc** [zwave-js/zwave-js-ui#3972] [Q&A] [answered]
such as:

![image](https://github.com/user-attachments/assets/580627ea-5067-400a-8a41-700ed871d4bc)

where the ip address is, say,  a raspberry pi with a zwave usb stick running ser2net?   or the [tubeszb zwave poe kit](https://tubeszb.com/product/z-wave-poe-kit/) ?  

thanks in advance

**Comment by kpine** [comment]
Yes, that's why the link tells you to use Z-Wave JS UI and input the `tcp://` serial port path. Is it not working?

https://zwave-js.github.io/zwave-js-ui/#/usage/setup?id=z-wave

> Serial port: The serial port where your controller is connected. This input allows to insert custom values, if your port is not listed or you are using tools like [ser2net](https://github.com/cminyard/ser2net) to make it available **throught TCP just write the port path and press enter**.

https://zwave-js.github.io/node-zwave-js/#/usage/tcp-connection?id=configuration-in-z-wave-js

> To use a remote serial port, enter **tcp://<hostname-or-ip>:<port> as the connection string in the "Serial Port" field of the Z-Wave JS configuration**

**Comment by ltgcc** [comment]
Hi!  thanks for the quick reply.  No, it's not working (and it may -totally- be user-oops)

The 'configuration contains errors.  fix it' message comes up right after I press 'save' after putting in the tcp://192.168.22.8:6638 (which is the ip of the tubeszb zwave appliance).

the docker container can see the tubeszb zwave appliance:

```
/home/zwave-js$ docker container  exec -it zwave-js-ui /bin/sh
/usr/src/app # nc -vz 192.168.22.8 6638
192.168.22.8 (192.168.22.8:6638) open
```

screenshot of the zwave-js-ui configuration page:




![image](https://github.com/user-attachments/assets/bea7140e-c6b6-4342-adc6-755923091e97)

Thanks!

**Reply by kpine** [reply, marked as answer]
Did you scroll down and fix the errors? They should be highlighted in red. For example, did you neglect to configure security keys?

https://zwave-js.github.io/zwave-js-ui/#/getting-started/quick-start?id=minimum-settings

![image](https://github.com/user-attachments/assets/40fe26cf-65ee-484b-87f8-205da708aebd)

**Reply by ltgcc** [reply]
ohhhhhhhh.   No, I didn't do that yet.   I usually do security and firewall and policy and selinux and all that kind of thing after everything has been working a while.

I'll do that now.

Thanks!

**Reply by ltgcc** [reply]
lol...yeah...minimum settings.  My oops.  learning the lesson yet again to read through the howto **first**.   For the 65535th time.  thanks.

Working it now.  I'll let you know.

**Reply by ltgcc** [reply]
ok.  Got past it.   configuration saved ok.   Thank you again.   this is resolved and I'll mark it as such.
